src/main/java/org/candlepin/subscriptions/tally/SnapshotSummaryProducer.java

```diff
@@ -91,11 +91,12 @@ private TallySummary createTallySummary(String accountNumber, List<TallySnapshot
 
     public void produceTallySummaryMessages(Map<String, List<TallySnapshot>> newAndUpdatedSnapshots) {
         AtomicInteger totalTallies = new AtomicInteger();
-        newAndUpdatedSnapshots.entrySet().stream()
-            .map(entry -> createTallySummary(entry.getKey(), entry.getValue())).forEach(tallySummary -> {
-                tallySummaryKafkaTemplate.send(tallySummaryTopic, tallySummary);
+        newAndUpdatedSnapshots.forEach((account, snapshots) -> snapshots.stream()
+            .map(snapshot -> createTallySummary(account, List.of(snapshot)))
+            .forEach(summary -> {
+                tallySummaryKafkaTemplate.send(tallySummaryTopic, summary);
                 totalTallies.getAndIncrement();
```

Comment:
Looks good.  Breaks up the snapshots by account and performs summaries.  Hopefully, that solves it.